### ISSUE TITLE
Получить всю активность пользователей при регистрации

### ISSUE THREAD

**Issue by fey** [Hexlet/hexlet-friends#323]
Согласно #173 теперь у нас пользователь в центре внимания, а не проекты. 
Что для этого нужно сделать
1. При регистрации пользователю устанавливаются настройки в гитхабе, чтобы пересылать информацию с гитхаба к нам на сервис. тут нужно подумать. как нужно сделать. Думаю можем из апишки подтягивать данные по пользователю. Плюсом пользователь наверное должен генерить апитокен, чтобы обращались с этим токеном к гитхабу
2. Нужна страница с настройками. Там пользователь может управлять своим аккаунтом на friends и имеет кнопку типа "синхронизировать", чтобы стянуть свежие данные.

**Comment by NevermoreKatana** [comment]
@fey Привет! Хотелось бы уточнить, какая информация по юзеру должна вытягиваться помимо той что уже есть при авторизации, какая цель в будущем у этой информации и что нужно сохранять в БД?
Я АПИшкой получил вот это, что из этого надо сохранять на сервере?
```json
 {
    "login": "NevermoreKatana",
    "id": 128977921,
    "node_id": "U_kgDOB7AMAQ",
    "avatar_url": "https://avatars.githubusercontent.com/u/128977921?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/NevermoreKatana",
    "html_url": "https://github.com/NevermoreKatana",
    "followers_url": "https://api.github.com/users/NevermoreKatana/followers",
    "following_url": "https://api.github.com/users/NevermoreKatana/following{/other_user}",
    "gists_url": "https://api.github.com/users/NevermoreKatana/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/NevermoreKatana/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/NevermoreKatana/subscriptions",
    "organizations_url": "https://api.github.com/users/NevermoreKatana/orgs",
    "repos_url": "https://api.github.com/users/NevermoreKatana/repos",
    "events_url": "https://api.github.com/users/NevermoreKatana/events{/privacy}",
    "received_events_url": "https://api.github.com/users/NevermoreKatana/received_events",
    "type": "User",
    "site_admin": false,
    "name": "Katana Nevermore",
    "company": "Not",
    "blog": "",
    "location": "Nevermore",
    "email": null,
    "hireable": null,
    "bio": "Hi, my name is Katana. I am not a professional developer from Russia, I am practically self-taught.",
    "twitter_username": null,
    "public_repos": 9,
    "public_gists": 0,
    "followers": 4,
    "following": 1,
    "created_at": "2023-03-26T12:45:16Z",
    "updated_at": "2023-09-14T10:57:30Z"
}
```

**Comment by fey** [comment]
@NevermoreKatana привет. 
Нам хорошо бы уметь получать инфу с гитхаба - в каких оргах состоит пользователь, какие у него репозитории, куда он контрибьютил. 
Задача hexlet-friends в том, чтобы все это собрать в себе, показать красивую инфу, типа портфолио.

**Comment by NevermoreKatana** [comment]
@fey 
Разве  hexlet-friends сейчас не делает этого? На скрине только часть кода, но там как я вижу пробегаются по всему что связано с юзером, если я правильно все понял. https://github.com/Hexlet/hexlet-friends/blob/main/contributors/utils/github_lib.py
Но нет такого представления в виде красивой инфы.
Т.е. остается переделать вью и фронт?

![image](https://github.com/Hexlet/hexlet-friends/assets/128977921/f44af377-cd29-4584-8b0c-599a31cb5a54)

**Comment by fey** [comment]
да, например у нас есть задача показзывать участников тех, которые относятся к одной оргинизации в гитхабе. 
Возможно сейчас мы собираем эту инфу. А возможно и нет. Это надо выяснить и добавить если нету.
ПЛюсом раньше как было - добавлялись новые репозитории и отслеживалась активность только тех, кто в этих репах участвует. 

К примеру я в сентябре создал несколько пулл реквестов, это видно в гитхабе. А в в сервисе их нет
![image](https://github.com/Hexlet/hexlet-friends/assets/697178/b547e086-fd2e-46be-97e7-8da8b0c46770)
![image](https://github.com/Hexlet/hexlet-friends/assets/697178/bd87f59c-2aab-437e-908c-74f2b5134760)

**Comment by TarakanovAndrey** [comment]
Сейчас разбираюсь с этой задачей. Правильно я понимаю, что сначала все стартовала от репозиториев и пользователи уже подтягивались с них. Сейчас, так как акцент на пользователе, то вектор смещается на регистрацию пользователся в системе и подтягивание информации уже начиная с юзера? И сейчас данные, загружаются частично. Мои ишьюсы не отобразились.

**Comment by fey** [comment]
да! Вы все верно поняли. Именно так раньше проект выглядел) Были проекты Хекслета, они были загружены, подгружались данные о юзерах только тех, что принимали участие в этих проектах. Потом акцент стал смещаться на пользователей. На их активность и тд. 

Насчет структуры бд - признаюсь, я в нее не смотрел глубоко. Да, это может поменяться. 

На мой взгляд щас не хватает кабинета с настройками пользователя, где он может например нажать кнопку "Обновить мою инфу", чтобы вся инфа о нем начала подгружаться.

**Comment by TarakanovAndrey** [comment]
> да! Вы все верно поняли. Именно так раньше проект выглядел) Были проекты Хекслета, они были загружены, подгружались данные о юзерах только тех, что принимали участие в этих проектах. Потом акцент стал смещаться на пользователей. На их активность и тд.
> 
> Насчет структуры бд - признаюсь, я в нее не смотрел глубоко. Да, это может поменяться.
> 
> На мой взгляд щас не хватает кабинета с настройками пользователя, где он может например нажать кнопку "Обновить мою инфу", чтобы вся инфа о нем начала подгружаться.

Да, с этим я сейчас и разбираюсь. И появилась проблема. Я не могу загрузить issues пользователя. Возвращается просто пустой списко и код 200. Хотя я с токеном авторизуюсь. При этом, например информацию по репам я получаю.

**Comment by TarakanovAndrey** [comment]
Я беру этот issue в работу

**Comment by fey** [comment]
@TarakanovAndrey как успехи?

**Comment by TarakanovAndrey** [comment]
Добрый день! Пишу, когда могу. Был в процессе поиска работы. А каждое тестовое это, как оказалось, свой стэк))) Поэтому, сейчас как немного все устаканится, отпишусь подробнее -- Андрей Тараканов8 988 488 92 48   04.03.2024, 18:54, "Nikolay Gagarinov" ***@***.***>***@***.*** как успехи?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by fey** [comment]
Хорошо, если что задачка все еще актуальная. Поэтому если кто-то еще хочет поучаствовать тут - вперед. Задача не совсем тривиальная и будет хорошо, если несколько людей попробую ее решить.

**Comment by serVmik** [comment]
Хочу поучавствовать. Решение данной задачи предпологает знание логики сайта, могу разобраться, но оптимальнее спросить). Посдкажите, с чего начать, чтоб приступить к решению данного вопроса. Может есть документация проекта?

**Comment by fey** [comment]
@serVmik сейчас у некоторых юзеров активность подтягивается, у каких-то нет. У нас есть команда, которая подтягивает данные с гитхаба. Можете с нее начать.
